code/pages/ExtensibleSearchPage.php

```diff
@@ -719,11 +748,76 @@ public function getSearchResults($data = null, $form = null) {
 
 		$startTime = microtime(true);
 
+		if($results = $this->getSearchEngineResults($data, $form)) {
+			$engine = $this->data()->SearchEngine;
+
+			return $this->customise($results)
+				->renderWith(array(
+					"{$engine}Search_results",
+					"{$engine}SearchPage_results",
+					'ExtensibleSearch_results',
+					'ExtensibleSearchPage_results',
+					'Page_results',
+					"{$engine}Search",
+					"{$engine}SearchPage",
+					'ExtensibleSearch',
+					'ExtensibleSearchPage',
+					'Page'
+				));
+		} else {
+
+			$results = $this->getSearchResultsFallback($data, $form);
+			// Render the full-text results using a listing template where defined.
+			if($this->data()->ListingTemplateID && $results) {
+				$template = DataObject::get_by_id('ListingTemplate', $this->data()->ListingTemplateID);
+				if($template && $template->exists()) {
+					$render = $this->data()->customise(array(
+						'Items' => $results
+					));
+					$viewer = SSViewer::fromString($template->ItemTemplate);
+					$results = $viewer->process($render);
+				}
+			}
+
+			// Render everything into the search page template.
+
+			$customisation = array(
+				'Results' => $results,
+				'Query' => $form ? $form->getSearchQuery() : null,
+				'Title' => _t('ExtensibleSearchPage.SearchResults', 'Search Results')
+			);
+			$output = $this->customise($customisation)->renderWith(array(
+				'ExtensibleSearch_results',
+				'ExtensibleSearchPage_results',
+				'Page_results',
+				'ExtensibleSearch',
+				'ExtensibleSearchPage',
+				'Page'
+			));
+			$totalTime = microtime(true) - $startTime;
+
+			// Log the details of a user search for analytics.
+
+			$this->service->logSearch(
+				$data['Search'],
+				$results ? count($results) : 0,
+				$totalTime,
+				'FallbackFullText',
```

Comment:
Although this is technically the full-text fallback, it will also be used when we're using full-text as the search engine. Please refer to my comments above :)